lib/app_revision.rb

```diff
@@ -0,0 +1,15 @@
+class AppRevision
+  def self.version
+    @version ||= begin
+      if revision_file.exist?
+        revision_file.read
```

Comment:
Checking a file's existence before reading has a subtle race condition. The file could be deleted by another process after line 4, before line 5.

Safer to try to read the file, and rescue the `Errno::ENOENT`:

```ruby
@version ||= begin
  revision_file.read
rescue Errno::ENOENT
  `git rev-parse HEAD`
end.strip
```

While it's not a bug we'll likely encounter in this case, the value in fixing it is to serve as a better example for other contributors copying the patterns they see here.